lib/iris/tests/unit/lazy_data/test_array_masked_to_nans.py

```diff
@@ -31,41 +31,45 @@
 
 
 class Test(tests.IrisTest):
-    def test_masked(self):
-        masked_array = ma.masked_array([[1.0, 2.0], [3.0, 4.0]],
-                                       mask=[[0, 1], [0, 0]])
-
-        result = array_masked_to_nans(masked_array).data
-
+    def _common_checks(self, result):
         self.assertIsInstance(result, np.ndarray)
         self.assertFalse(isinstance(result, ma.MaskedArray))
         self.assertFalse(ma.is_masked(result))
 
+    def test_masked_input(self):
+        masked_array = ma.masked_array([[1.0, 2.0], [3.0, 4.0]],
+                                       mask=[[0, 1], [0, 0]])
+
+        result = array_masked_to_nans(masked_array)
+
+        self._common_checks(result)
         self.assertArrayAllClose(np.isnan(result),
                                  [[False, True], [False, False]])
         result[0, 1] = 777.7
         self.assertArrayAllClose(result, [[1.0, 777.7], [3.0, 4.0]])
 
+    def test_unmasked_input(self):
+        unmasked_array = np.array([1.0, 2.0])
+        result = array_masked_to_nans(unmasked_array)
+        # Non-masked array is returned as-is, without copying.
+        self.assertIs(result, unmasked_array)
+
     def test_empty_mask(self):
```

Comment:
Masked arrays being the horrible things they are, it might be worth checking also the case where the 'mask' keyword is not set (the actual default is  "mask=np.ma.nomask").
For a ghastly distinction, this is not the same thing as `mask=zeros(...)` or `mask=None` or `mask=False`, all of which allocate an actual (empty) mask _array_: 
```
>>> print np.ma.masked_array([1, 2], mask=None).mask
[False False]
>>> print np.ma.masked_array([1, 2]).mask
False
>>> print np.ma.masked_array([1, 2], mask=np.ma.nomask).mask
False
>>> 
```
Urrrgh! Don't get me started...